images/build/debian-iptables/distroless/get-binaries-from-package.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# USAGE: get-binaries-from-package.sh /opt/stage package1 package2
+# TODO: The above is not true. I will fix it!
+#
+# Stages all the packages and its dependencies (+ libraries and copyrights) to $1
+#
+# This is intended to be used in a multi-stage docker build with a distroless/base
+# or distroless/cc image.
+
+. package-utils.sh || exit 1 
+
+get_file_list() {
+    IFS="
+    "
+    REQUIRED_FILES="$(dpkg -L "${1}" | grep -vE '(/\.|/s?bin/|/usr/share/(man|doc|.*-completion))' | sed 's/\n/ /g')"
+    for file in $REQUIRED_FILES; do 
+        if [ -f "$file" ]; then
+            stage_file "${file}" "${STAGE_DIR}"
+        fi
+    done
+
+    BIN_LIST="$(dpkg -L "${1}" | grep -E '/s?bin/' |sed 's/\n/ /g')"
+    for binary in $BIN_LIST; do
+        /stage-binary-and-deps.sh "${binary}" "${2}"
+    done
+}
+
+get_dependent_packages() {
+    apt-cache depends "${1}" |grep Depends|awk -F '.*Depends:[[:space:]]?' '{print $2}'
```

Comment:
I'm not a Debian person but I'm guessing there's an easier way to do this...

Reply:
There is not, not that I could find. You either need a package staged (like dpkg -I package.deb) or using apt-cache depends.

Probably this AWK thing is too much, but I couldn't figure out a way of getting like a json/xml/whatever metadata with a package dependency

Reply:
@ehashman you are a debian person, do you know if there is an easier way?

Reply:
Can we agree this is a doable approach now and clean-up as soon as some debian expert appears? :D